src/views/Profile/PersonalInfo/PersonalInfo.tsx

```diff
@@ -188,8 +202,20 @@ export const PersonalInfo: React.FC<IEditAuthorProps> = ({
     setVisitFields({ ...visitFields, [inputIdentifier]: true });
   };
 
-  const buttonClickHandler = () => {
-    // TODO dispatch new author to server
+  const buttonClickHandler = async (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    e.preventDefault();
+    setIsLoading(true);
+    const response = author
+      ? await updateAuthorById({ authorId: author.id, ...newAuthorValues })
+      : await createAuthor(newAuthorValues);
```

Comment:
may `createAuthor` or `updateAuthorById` throw? If yes, I propose to add `try .. catch .. finally` to ensure `setIsLoading(false)` will happen anyway

Reply:
yes, i will add it

> may `createAuthor` or `updateAuthorById` throw? If yes, I propose to add `try .. catch .. finally` to ensure `setIsLoading(false)` will happen anyway